docs/DafnyRef/UserGuide.md

```diff
@@ -103,7 +103,43 @@ on the [Dafny INSTALL page in the wiki](https://github.com/dafny-lang/dafny/wiki
 
 ## 24.6. The Dafny Server
 
-TO BE WRITTEN
+Before Dafny [implemented](https://github.com/dafny-lang/dafny/tree/master/Source/DafnyLanguageServer) the official [Language Server Protocol](https://microsoft.github.io/language-server-protocol/), it implemented its own protocol to offer the [emacs mode](https://github.com/boogie-org/boogie-friends), which resulted in a project called [DafnyServer](https://github.com/dafny-lang/dafny/tree/master/Source/DafnyServer).
+
+The Dafny Server has [integration tests](https://github.com/dafny-lang/dafny/tree/master/Test/server) that serve as the basis of the documentation.
+
+The executable DafnyServer is a simple read-eval loop. For example, if you compile and run `DafnyServer.exe`, you could paste the following command:
+
+```dafny
+verify
+eyJhcmdzIjpbIi9jb21waWxlOjAiLCIvcHJpbnRUb29sdGlwcyIsIi90aW1lTGltaXQ6MjAiXSwi
+ZmlsZW5hbWUiOiJ0cmFuc2NyaXB0Iiwic291cmNlIjoibWV0aG9kIEEoYTppbnQpIHJldHVybnMg
+KGI6IGludCkge1xuICBiIDo9IGE7XG4gIGFzc2VydCBmYWxzZTtcbn1cbiIsInNvdXJjZUlzRmls
+ZSI6ZmFsc2V9
+[[DAFNY-CLIENT: EOM]]
+```
+
+The interpreter sees the command `verify`, and then starts reading every line until it sees `[[DAFNY-CLIENT: EOM]]`
+The payload is a base64 encoded string that you could encode or decode using JavaScript's `atob` and `btoa` function.
```

Comment:
Why mention javascript here?

Reply:
Because that's why I used to understand what these server commands were, it's something everyone can test with.